utils/cliutils/utils.go

```diff
@@ -401,19 +401,32 @@ func handleSecretInput(c *cli.Context, stringFlag, stdinFlag string) (secret str
 	return commonCliUtils.HandleSecretInput(stringFlag, c.String(stringFlag), stdinFlag, c.Bool(stdinFlag))
 }
 
-func GetSpec(c *cli.Context, isDownload bool) (specFiles *speccore.SpecFiles, err error) {
+func GetSpec(c *cli.Context, isDownload, overrideFieldsIfSet bool) (specFiles *speccore.SpecFiles, err error) {
 	specFiles, err = speccore.CreateSpecFromFile(c.String("spec"), coreutils.SpecVarsStringToMap(c.String("spec-vars")))
 	if err != nil {
 		return nil, err
 	}
-	// Override spec with CLI options
+	trimPatternPrefixIfDownload(specFiles, isDownload)
+	overrideSpecFieldsIfNeeded(c, specFiles, overrideFieldsIfSet)
```

Comment:
I find this a bit more simple to the reader:
```suggestion
	if isDownload {
		trimPatternPrefix(specFiles)
	}
	if overrideFieldsIfSet {
		overrideSpecFields(c, specFiles)
	}
```